example_sdk_client/optimizer.py

```diff
@@ -26,6 +28,7 @@ def handle_on_finished(job: Job, result: JobResult):
         f"Status: {result.result_type}, output esdl length: {len(result.output_esdl)}, "
         f"logs length: {len(result.logs)}"
     )
+    STOP_EVENT.set()
```

Comment:
@lfse-slafleur Could this STOP_EVENT thread cause some cleanup steps to be skipped? For instance, deleting the job result and status queues after the job is finished and the job result returns to the client.

Reply:
Multithreading is a difficult subject so I am tempted to never say never in these situations xD. However, setting the STOP_EVENT should signal that the main thread is allowed to proceed and will call omotes_if.stop() soon. This will lead to function https://github.com/Project-OMOTES/omotes-sdk-python/blob/main/src/omotes_sdk/internal/common/broker_interface.py#L546 and https://github.com/Project-OMOTES/omotes-sdk-python/blob/main/src/omotes_sdk/internal/common/broker_interface.py#L401 . This will cause the eventloop to eventually run a `.cancel` on all subscriptions, close the channel and close the connection.

Within the same eventloop the task is scheduled to eventually run https://github.com/Project-OMOTES/omotes-sdk-python/blob/main/src/omotes_sdk/omotes_interface.py#L172 .

These are in parallel in a 'race' so depening on scheduling, it would appear that it is possible that allowing the script to stop as soon as a result is received may cause the progress and job status queues to remain.

I didn't really dig into it because it is an example script.

Reply:
I compared the original `time.sleep()` with `STOP_EVENT` implementations and run `optimizer.py`. From what I saw in the logs and RabbitMQ interface. The first one does clean up job queue**s** properly, while the second one with `STOP_EVENT.set()` seems to signal the connection close too early and only cleans up the job progress queue (instead of the job result queue? why?). 

With `time.sleep()`
![image](https://github.com/user-attachments/assets/e9b4b831-dd47-44bf-bfb7-6711f58935bb)

With `STOP_EVENT`
![image](https://github.com/user-attachments/assets/bf857106-79a5-4e94-930d-ca7685d4914c)

And how are the clients (e.g. OMOTES REST) implementing job submission in practice? Also with `STOP_EVENT` implementation? As it seems to me that it could cause many leftover job queues in the Rabbitmq.

Reply:
@cwang39403 

`why?` <-- See my previous comment. There is no guarantee that https://github.com/Project-OMOTES/omotes-sdk-python/blob/main/src/omotes_sdk/omotes_interface.py#L172 is run before the client is stopped.

`And how are the clients (e.g. OMOTES REST) implementing job submission in practice?` The component stays online indefinitely. So there is no equal to STOP_EVENT except due to a manual stop. Then, it is still theoratically possible that the progress & status queues remain but that should happen very rarily.